--- js/rubik.js	Mon Mar 28 15:35:00 2011
+++ js/rubik.js	Mon Mar 28 15:42:30 2011
@@ -2,21 +2,7 @@
  * Implementation of Drupal behavior.
  */
 Drupal.behaviors.rubik = function(context) {
-  // If there are both main column and side column buttons, only show the main
-  // column buttons if the user scrolls past the ones to the side.
-  $('div.form:has(div.column-main div.buttons):not(.rubik-processed)').each(function() {
-    var form = $(this);
-    var offset = $('div.column-side div.buttons', form).height() + $('div.column-side div.buttons', form).offset().top;
-    $(window).scroll(function () {
-      if ($(this).scrollTop() > offset) {
-        $('div.column-main div.buttons', form).show();
-      }
-      else {
-        $('div.column-main div.buttons', form).hide();
-      }
-    });
-    form.addClass('rubik-processed');
-  });
+  // the form button behavior is silly
 
   $('a.toggler:not(.rubik-processed)', context).each(function() {
     var id = $(this).attr('href').split('#')[1];
